src/newscale/interfaces.py

```diff
@@ -54,40 +62,116 @@ def read(self, address: str = None):
         raise NotImplementedError("To be implemented by the child class.")
 
 
+class NewScaleSerial():
```

Comment:
Unless we are explicitly trying to support Python 2, let's stick with `class NewScaleSerial:` instead of `class NewScaleSerial():` since the both inherit from `object`.

[lengthy discussion on the matter](https://stackoverflow.com/a/45062077/3312269)